pkg/cli/snapshot_helm.go

```diff
@@ -44,8 +50,62 @@ func Snapshot(ctx context.Context, args []string, globalFlags *flags.GlobalFlags
 		}
 	}
 
-	// run snapshot pod
-	return pod.RunSnapshotPod(ctx, restConfig, kubeClient, []string{"/vcluster", "snapshot"}, vCluster, podOptions, snapshotOpts, log)
+	if !async {
+		// run snapshot pod
+		return pod.RunSnapshotPod(ctx, restConfig, kubeClient, []string{"/vcluster", "snapshot"}, vCluster, podOptions, snapshotOpts, log)
+	}
+
+	// get virtual kube config & client
+	vKubeConfig, err := clihelper.GetVClusterKubeConfig(ctx, restConfig, kubeClient, vCluster, log)
+	if err != nil {
+		return fmt.Errorf("failed to get virtual cluster config: %w", err)
+	}
+	virtualHTTPClient, err := rest.HTTPClientFor(vKubeConfig)
+	if err != nil {
+		return fmt.Errorf("failed to get http client: %w", err)
+	}
+	requestURL, err := url.JoinPath(vKubeConfig.Host, routes.PostSnapshotsURL)
+	if err != nil {
+		return fmt.Errorf("failed to join url: %w", err)
+	}
+
+	// create snapshot request ConfigMap
+	snapshotRequest := &snapshot.Request{
+		Spec: snapshot.RequestSpec{},
+	}
+	snapshotRequestData := map[string]interface{}{
+		snapshot.RequestKey: snapshotRequest,
+		snapshot.OptionsKey: snapshotOpts,
+	}
+	snapshotRequestDataJSON, err := json.Marshal(snapshotRequestData)
+	if err != nil {
+		return fmt.Errorf("failed to marshal snapshot request data: %w", err)
+	}
+	httpRequest, err := http.NewRequest(http.MethodPost, requestURL, bytes.NewReader(snapshotRequestDataJSON))
+	if err != nil {
+		return fmt.Errorf("failed to create http request: %w", err)
+	}
+	response, err := virtualHTTPClient.Do(httpRequest)
```

Comment:
This will fail for any virtual cluster below v0.29, correct?
Just wondering if there's a way to provide a better UX here (so either checking a virtual cluster version, and for `<v0.29` avoid all these operations and just output "async snapshots not supported for this vCluster version, please upgrade" or if we don't have a way to know vCluster version here, have a special handling for an error returned by pre-v0.29 (I guess the `/vcluster/snapshots` will return 404 in this case, right?)